File: Ejercicio 7/salud.py
class Paciente:
    def __init__(self, nombre):
        self.nombre = nombre


class Medico:
    def __init__(self, nombre):
        self.nombre = nombre

    def atender(self, paciente):
        return f"Atiende a {paciente.nombre}"


class Cita:
    def __init__(self, paciente, medico):
        self.paciente = paciente
        self.medico = medico

    def resumen(self):
        return f"Cita: {self.medico.nombre} con {self.paciente.nombre}"



p = Paciente("Juan Alberto")
m = Medico("Dr. Carlos Cruz")
cita = Cita(p, m)

print(cita.resumen())
print(m.atender(p))

File: Ejercicio 7/salud.java
class Paciente {
    String nombre;

    public Paciente(String nombre) {
        this.nombre = nombre;
    }
}

class Medico {
    String nombre;

    public Medico(String nombre) {
        this.nombre = nombre;
    }

    public String atender(Paciente p) {
        return "Atiende a " + p.nombre;
    }
}

class Cita {
    Paciente paciente;
    Medico medico;

    public Cita(Paciente p, Medico m) {
        this.paciente = p;
        this.medico = m;
    }

    public String resumen() {
        return "Cita: " + medico.nombre + " con " + paciente.nombre;
    }
}

public class Main {
    public static void main(String[] args) {
        Paciente p = new Paciente("Juan Alberto");
        Medico m = new Medico("Dr. Carlos Cruz");
        Cita c = new Cita(p, m);

        System.out.println(c.resumen());
        System.out.println(m.atender(p));
    }
}
